python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -944,41 +972,119 @@ def _get_entities_from_payload(self, data):
             that order.
```

Comment:
It's not clear to me based on the documentation and the code what the payload can be. There seems to be a lot of edge cases in the method and I think it would be nice to provide an example of each type on payloads you can get from the browser.

Reply:
I'm getting a feeling that each entity has the field `id` and `type` and sometimes `project`, because the web app has been inconsistent about this in the past maybe? Is that the right interpretation?

Reply:
In any case, clearing that up with a comment would be helpful.

Reply:
Will do, and yeah, it's a bit squirrelly and inconsistent. What I was fixing in this PR is yet another inconsistency that comes from one specific section of the My Tasks page.

Reply:
It's worth noting that most of the inconsistencies I would consider bugs in the toolkit menu code in the web app. Especially this one here from My Tasks. However, it's significantly easier to work around it in Python than it is to hunt it down in javascript, and it's safer to release, as well.